sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java

```diff
@@ -547,6 +561,28 @@ interface WithHns {
             WithCreate withHnsEnabled(boolean enabled);
         }
 
+        /** The stage of the storage account definition allowing to specify whether cross tenant replication is enabled. */
+        interface WithAllowCrossTenantReplication {
+            /**
+             * Specifies whether cross tenant replication will be enabled or not.
+             *
+             * @param enabled whether Hns will be enabled or not
+             * @return the next stage of storage account definition
+             */
+            WithCreate withAllowCrossTenantReplication(boolean enabled);
+        }
+
+        /** The stage of the storage account definition allowing to specify whether default to oauth authentication is enabled. */
+        interface  WithDefaultToOAuthAuthentication {
+            /**
+             * Specifies whether default to oauth authentication will be enabled or not.
+             *
+             * @param enabled whether default to oauth authentication will be enabled or not
+             * @return the next stage of storage account definition
+             */
+            WithCreate withDefaultToOAuthAuthentication(boolean enabled);
```

Comment:
Same, maybe `enableDefaultToOAuthAuthentication()`.

Reply:
That means you need to use the "enable" and "disable" methods to enable and disable "allowCrossTenantReplication" and "defaultToOAuthAuthentication" respectively, is that right?

Reply:
Yes. Depends on the meaning of `null`.

Reply:
Yes, if default is false, then in create phase, we only need `enableCrossTenantReplication()`(or `enableDefaultToOAuthAuthentication()`).

In update phase, we probably need both `enableCrossTenantReplication()` and `disableCrossTenantReplication()`.

Reply:
Fixed in the new version.